dependencies: Add black to development dependencies

justwheel opened the pull request:
This commit adds Black, a Python code formatter, to our development
dependencies in the Pipfile:

https://github.com/psf/black

This is a useful tool for inspecting code quality and improving
readability. This commit is adding it to the dependencies, even though
nothing is actually using it just yet. I'll probably spend more time on
this in the second half of Sprint 5.

justwheel committed:
dependencies: Add black to development dependencies

This commit adds Black, a Python code formatter, to our development
dependencies in the Pipfile:

    https://github.com/psf/black

This is a useful tool for inspecting code quality and improving
readability. This commit is adding it to the dependencies, even though
nothing is actually using it just yet. I'll probably spend more time on
this in the second half of Sprint 5.

Signed-off-by: Justin W. Flory <git@jwf.io>

nlMeminger commented:
If we aren't using it yet, why are we adding it now?

jwflory commented:
@nlMeminger I wanted to get this change in earlier since it requires rebuilding the Docker image (i.e. dependency change). It gives more time for `black` to get into everyone's container builds before I actually start working with it next week and things would break.

nlMeminger commented:
Regardless of when it’s added, everyone will have to go through a full
container rebuild eventually. I’d rather keep dependencies out until
they’re actually used in case things change and it doesn’t actually get
used.

On Thu, Nov 7, 2019 at 08:54 Justin W. Flory <notifications@github.com>
wrote:

> @nlMeminger <https://github.com/nlMeminger> I wanted to get this change
> in earlier since it requires rebuilding the Docker image (i.e. dependency
> change). It gives more time for black to get into everyone's container
> builds before I actually start working with it next week and things would
> break.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/jwflory/django-rit-grasa/pull/164?email_source=notifications&email_token=AELFBEOGMSIT5LUDNH2I35TQSQMY3A5CNFSM4JJ6CKT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEDMPDUA#issuecomment-551088592>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AELFBEKWTNWQDN2NIX6QTN3QSQMY3ANCNFSM4JJ6CKTQ>
> .
>

jwflory commented:
@nlMeminger That's fair feedback. I'll write up an issue with my plan of action first and follow back up here soon.

justwheel committed:
Add shell script to lint code using Black

This commit adds a basic shell script to lint our code using the Black
code formatter. I confirmed this works when Black is installed as a
dependency in my Pipenv-managed virtualenv.

Signed-off-by: Justin W. Flory <git@jwf.io>
